grakn-graph/src/main/java/ai/grakn/graph/internal/RelationTypeImpl.java

```diff
@@ -155,8 +155,38 @@ public void delete(){
 
     @Override
     void trackRolePlayers(){
-        //TODO this operation should not call reify() rather it should call reified. Need to think of a way to validate Relation Type changes when the instances of the relation type are not reified
-        instances().forEach(concept -> ((RelationImpl) concept).reify().castingsInstance().forEach(
-                rolePlayer -> vertex().graph().txCache().trackForValidation(rolePlayer)));
+        instances().forEach(concept -> {
+            RelationImpl relation = RelationImpl.from(concept);
+            if(relation.reified().isPresent()){
+                relation.reified().get().castingsRelation().forEach(rolePlayer -> vertex().graph().txCache().trackForValidation(rolePlayer));
+            }
+        });
+    }
+
+    @Override
+    public Collection<Relation> directInstances(){
+        Collection<Relation> instances = super.directInstances();
+
+        if(isImplicit()){
+            //If the relation type is implicit then we need to get any relation edges it may have.
+            //Unfortunately this is a slow process
+
+            relates().stream().
+                    flatMap(role -> role.playedByTypes().stream()).
+                    forEach(type ->{
+                        //Traversal is used here to take advantage of vertex centric index
+                        vertex().graph().getTinkerTraversal().V().
```

Comment:
I am really not happy with this one. We need elastic search or a more clever way to handle these situations.